Add making-bitcoin-more-private-with-cisa review to bitcoin-magazine/bitcoin-2024 (#548)
bitcointranscripts/bitcointranscripts: bitcoin-magazine/bitcoin-2024/making-bitcoin-more-private-with-cisa.md

---
title: 'Making Bitcoin More Private with CISA'
transcript_by: 'markon1-a via review.btctranscripts.com'
media: 'https://www.youtube.com/watch?v=HvI7NPI_Pk0'
date: '2024-08-29'
tags:
  - 'cisa'
speakers:
  - 'Fabian Jahr'
  - 'Jameson Lopp'
  - 'Craig Raw'
summary: 'Delve into the world of Cross-Input Signature Aggregation (CISA) with our expert panel from Sparrow Wallet, Bitcoin Core, and Casa. This discussion illuminates how CISA could revolutionize Bitcoin transactions by enhancing privacy and reducing fees. Uncover the technical aspects of half and full signature aggregation and understand their potential impact on the Bitcoin network.'
---
Fabian Jahr: 00:00:01

So there's one person on the stage here that is not up there.
Why?

> Nifi: 00:00:07

Reviewer: This is "Lisa Neigut" aka Nifty, so sometimes you will hear calling her Nifty

Author: This I couldn't figure out.

So if panelists, you could tell the audience about the project that you're working on and what your first, like where you first heard about CISA from, if that makes sense.
So Craig, do you want to start?

Craig Raw: 00:00:55

So I built Sparrow Wallet.
It's a security and privacy focused wallet.
The first time I heard about CISA was really from you know other privacy activists in the Bitcoin space who were talking about how they were really hoping that this cross-input signature aggregation was going to be shipped as part of the Taproot upgrade.
Obviously, we know that that didn't happen so when you know you're looking at things from a privacy point of view you want to do things like create multi-party transactions and as we will hear, cross-input signature aggregation really provides an interesting basis for being able to do that with more economic benefits.
So that was the first time that I kind of really started to look at it was from the privacy angle.

Fabian Jahr: 00:01:54

Yeah, I primarily work on Bitcoin Core and I can't really remember a specific time when I heard about it for the first time.

> Like the time between the SegWit Softfork and the Taproot Softfork was really when I got deeper into Bitcoin and like started contributing to Bitcoin Core and somehow it was always there.

Reviewer: Here you write soft fork as "Softfork" and later as "softfork", but on other occasions you keep it as "soft fork" or "soft-fork". I know that we still don't point to a specific style guide but you should at least have a single way of referring to it in a transcript that you are editing.

Please keep "soft fork" as specified here: https://github.com/bitcoinops/bitcoinops.github.io/blob/master/STYLE.md

## Incentivizing Participation in Coin Joins

Basically, you run into some problems where if you're receiving a lot of transactions, you're basically creating a lot of unspent transaction outputs in your wallet, it becomes problematic if at some point in the far future you want to go spend them and perhaps the fee rates in the market for block space have gone up a lot, it can become insanely expensive to spend your own money and this inevitably catches a lot of people by surprise if they haven't been through a full market cycle before.
So I think that this is another potential interesting aspect of aggregation is that once again we would be pushing the balance forward a bit to help incentivize people to you know clean up their UTXOs because we're not penalizing them as much by making it really expensive to do so.

Fabian Jahr: 00:10:37

> And maybe to expand on that, so there is a financial incentive then for people to participate in Coin joins and that gives them additional privacy.

Reviewer: ```suggestion
And maybe to expand on that, so there is a financial incentive then for people to participate in coinjoins and that gives them additional privacy.
```

And of course, we have not yet taken advantage of that.

Speaker 2: 00:16:33
Craig Raw: 00:16:33

Yeah, so we're in a very different space from that now.
I think something to bear in mind, if you're thinking about it from that point of view, is that, and I heard Brandon Black actually, who was here earlier today talking about Opcad, is that I think Caesar is probably the most low-risk soft fork that we could imagine.
And I'm talking here about the well-understood half signature aggregate form of it.
I think something to bear in mind, if you're thinking about it from that point of view, is that, and I heard Brandon Black actually, who was here earlier today talking about Opcad, is that I think CISA is probably the most low-risk soft fork that we could imagine.
And I'm talking here about the well-understood half-signature aggregate form of it.
You know, it's really from a security model point of view, the cryptographers tell us that there is no risk effectively or very low risk to doing this.
It really takes very well understood properties of Schnorr, which is really just adding these signatures up and uses that.

## The Limited Potential of Signature Compression

> Speaker 2: 00:17:18

Reviewer: I'm keep repeating that you shouldn't remove speaker attribution after a chapter name

Author: I had this in mind but I think I did the opposite.

> So as far as I'm aware off the top of my head, there's a pretty limited amount of adoption of Schnorr so far because I think Musig 2, is the main production-ready standard that people are using.

Reviewer: ```suggestion
So as far as I'm aware off the top of my head, there's a pretty limited amount of adoption of Schnorr so far because I think MuSig2, is the main production-ready standard that people are using.
```

Author: I was to change these just forgot after edits.

> And that can only do an N of N, like a two of two or a three of three type of scheme, so it's kind of limited in its flexibility.

Reviewer: for signatures we follow this notation
```suggestion
And that can only do an N-of-N, like a 2-of-2 or a 3-of-3 type of scheme, so it's kind of limited in its flexibility.
```

Author: Okay.

I care about this a lot, operating a multi-signature self-custody wallet.

> We're waiting for the further evolution of threshold signature schemes like Frost that can do more arbitrary, K of N type of multi-sig.

Reviewer: Signatures notation and also FROST
```suggestion
We're waiting for the further evolution of threshold signature schemes like FROST that can do more arbitrary, K of N type of multi-sig.
```

So for the half-arc, there's a BIP draft, but it's not a pull request on BIP repository, but there's just minor to-dos like test vectors there that are open however for the full aggregation we still need to develop the signature scheme, so that requires quite a bit of effort from people reversed in cryptography and specifically on the side of the interactivity.
We will also want to have a security proof for that scheme that comes out on the other hand and that means people with very specific talents and an experience in that area will also be needed in process.
These I think are the big to-dos that we have.
The people that can do this stuff need to come together and work on it.
Personally, I spend quite a bit of time on it right now.
But maybe something to everyone watching this or in the audience, if you want to get involved, let us know.

Jameson Lopp: 00:21:01

Call your local developer and demand signature aggregation today.

Craig Raw: 00:21:05

Yeah, and just to that point, Fabian has developed a great site where you can learn more.
So, if the concepts that we've talked about here today make sense, if you want to understand them better.

> If you go to sizeresearch.org, that's C-I-S-A research dot org, you can really learn much more about it.

Reviewer: The domain was picked wrong by the AI transcription, but then it spells it letter by letter. This is something that we expect the reviewers to fix and honestly @markon1-a this is a signal that you are still not paying attention to the full context of the transcript.
```suggestion
If you go to <cisaresearch.org> you can really learn much more about it.
```

Author: Yes this was an obvious one I didn't concentrate on.
